Begin.py/begin.py

```diff
@@ -52,76 +49,85 @@ def __repr__(self):
+                    van het bestand filename. Type: 1 hele lange string.
         """
 
-        with open("/home/annemarleen/programming/CS_for_all/Eindopdracht/test.txt") as f:
-            self.text = f.read().replace("\n", "").rstrip("")            
+        file = open(filename)
+        text = file.read().replace("\n", "").rstrip("")
+
+        return text
 
-        return self.text
-        
-    
-  
     def make_sentence_lengths(self, text):
-        """        
+        """
         De make_sentence_lengths(self) moet parameter text (type string) gebruiken om de
         dictionary self.sentence_lengths te vullen.
         output: dictionary self.sentence_lengths.
-        Het resultaat van deze functie is: dictionary: sentence; words 
+        Het resultaat van deze functie is: dictionary: sentence; words
 
         assert tm.sentence_lengths == {16: 1, 5: 1, 6: 1} => 1 zin met 16 woorden, 1 zin met 5 woorden, 1 zin met 6 woorden
-        
+
         """
         # dus, je moet het aantal zinnen tellen, én het aantal woorden in de zin.
-        # als x is zin, return True. If True, count words. if word endswhith '.?!' dan 
+        # als x is zin, return True. If True, count words. if word endswhith '.?!' dan
         # count_words. If word endswith '.?!' dan return is_sentence. count_sentence =+ 1
         # het resultaat hiervan wordt dmv een dictionary getoond.
 
-        print("The whole original string is:\n",self.text)
+        print("The whole original string is:\n", text)
         print("")
-        
-        pw = "$"
-        LoW = self.text.split()
-        count = 0
-        zin = []
-
-        #print(LoW)         
-
-        for nw in LoW:
-            if nw not in ".?!" or pw == "$":
-                count +=1
-            if nw[-1] in ".?!":
-              #  pw = "$" 
-                zin += [count]
-                count = 0
+
+        previous_word = "$"
+        words = text.split()
+        sentences = []
+        word_length = 0
+
+        # print(LoW)
+
+        for word in words:
+            if word not in ".?!" or previous_word == "$":
+                word_length += 1
+            if word[-1] in ".?!":
+                sentences += [word_length]
+                word_length = 0
             else:
-                pw=nw
+                previous_word = word
 
-        for number in zin:
-            if number not in self.sentence_lengths: 
+        for number in sentences:
+            if number not in self.sentence_lengths:
                 self.sentence_lengths[number] = 1
-            else: 
+            else:
                 self.sentence_lengths[number] += 1
 
-        return self.sentence_lengths  
+        return self.sentence_lengths
+
+    # _TESTS
+
+
+def run():
+    model = TextModel()
+
+    # Toon model voor het vullen
+    print('TextModel:\n', model)
+    # de text ophalen
+    text = model.read_text_from_file("../test.txt")
```

Comment:
dit is een dynamisch file path, dan heb je geen gebruiker specifiek pad meer.
../ is een map terug.